pkg/runner/runner.go

```diff
@@ -98,6 +110,21 @@ func Run(cfg *Config) error {
 	log.Info("msg", "Starting up...")
 	log.Info("msg", "Listening", "addr", cfg.ListenAddr)
 
+	telemetryEngine, err := telemetry.NewEngine(client.Connection, PromscaleID)
+	if err != nil {
+		log.Error("msg", "aborting startup due to error in setting up telemetry-engine", "err", err.Error())
+		return fmt.Errorf("creating telemetry-engine: %w", err)
+	}
+
+	telemetryEngine.Start()
+	defer telemetryEngine.Stop()
+
```

Comment:
is it safer to start the engine after the `err = api.RegisterMetricsForTelemetry(telemetryEngine)`call?

Reply:
Yes, there should not be any problem as we are using concurrency safe `sync.Map`.